src/utils/getPreparedPeople.tsx

```diff
@@ -0,0 +1,59 @@
+import { Person } from '../types';
+import { SortBy } from '../types/SortBy';
+
+export const getPreparedPeople = (
+  people: Person[],
+  sex: string | null,
+  query: string | null,
+  centuries: string[],
+  sortBy: string | null,
+  sortOrder: string | null,
+) => {
+  let preparedPeople = [...people];
+
+  if (sex) {
+    preparedPeople = preparedPeople.filter(person => person.sex === sex);
+  }
+
+  if (query) {
+    preparedPeople = preparedPeople.filter(
+      person => person.fatherName?.toLowerCase().trim()
+        .includes(query.toLowerCase().trim())
+        || person.motherName?.toLowerCase().trim()
+          .includes(query.toLowerCase().trim())
+        || person.name.toLowerCase()
+          .includes(query.toLowerCase().trim()),
+    );
+  }
+
+  if (centuries.length) {
+    preparedPeople = preparedPeople.filter(
+      person => centuries.includes(
+        Math.ceil(person.born / 100).toString(),
+      ),
+    );
+  }
+
+  if (sortBy) {
+    preparedPeople.sort((a, b) => {
+      switch (sortBy) {
+        case SortBy.name:
```

Comment:
It's ok for this task, but the function looks a bit complicated, you can split logic into two functions. The first one is for filtering, and the second is for sorting